internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/controller/TestVeniceHelixAdminWithIsolatedEnvironment.java

```diff
@@ -304,6 +309,93 @@ public void testIdempotentStoreDeletion() {
     Assert.assertNull(storeConfigAccessor.getStoreConfig(newStoreName));
   }
 
+  @Test
+  public void testStoreLifecycle_ValidateDeleted() {
+    String storeName = Utils.getUniqueString("test_validate_deleted_store");
+    int largestUsedVersion = Store.IGNORE_VERSION;
+
+    // Create store
+    veniceAdmin.createStore(clusterName, storeName, storeOwner, KEY_SCHEMA, VALUE_SCHEMA);
+
+    // Disable reads and writes before deletion (required by Venice)
+    veniceAdmin
+        .updateStore(clusterName, storeName, new UpdateStoreQueryParams().setEnableReads(false).setEnableWrites(false));
+
+    // Delete store
+    veniceAdmin.deleteStore(clusterName, storeName, largestUsedVersion, true);
+
+    // Validate deletion
+    StoreDeletedValidation result = veniceAdmin.validateStoreDeleted(clusterName, storeName);
+
+    Assert.assertTrue(result.isDeleted(), "Store should be completely deleted. Failure reason: " + result.getError());
+  }
+
+  @Test
+  public void testStoreLifecycle_ValidateDeleted_WithLeftoverTopics() {
+    String storeName = Utils.getUniqueString("test_validate_deleted_store_with_leftover_topics");
+
+    // Create store
+    veniceAdmin.createStore(clusterName, storeName, storeOwner, KEY_SCHEMA, VALUE_SCHEMA);
+
+    // Create a version topic manually to simulate a version that existed
+    String versionTopicName = Version.composeKafkaTopic(storeName, 1);
+    PubSubTopic versionTopic = pubSubTopicRepository.getTopic(versionTopicName);
+    TopicManager topicManager = veniceAdmin.getTopicManager();
+
+    // Create the topic to simulate that a version existed
+    topicManager.createTopic(versionTopic, 1, 1, true);
+
+    // Verify topic exists
+    Assert.assertTrue(topicManager.containsTopic(versionTopic), "Version topic should exist");
+
+    // Disable reads and writes before deletion (required by Venice)
+    veniceAdmin
+        .updateStore(clusterName, storeName, new UpdateStoreQueryParams().setEnableReads(false).setEnableWrites(false));
+
+    // Delete store metadata only (without trying to delete versions through normal flow)
+    // This simulates the scenario where store metadata is deleted but topics remain due to disabled cleanup
+    try {
+      veniceAdmin.getHelixVeniceClusterResources(clusterName).getStoreMetadataRepository().deleteStore(storeName);
+    } catch (Exception e) {
```

Comment:
Using try-catch blocks in tests is generally not a good idea. If an exception is expected, it should be handled using expectThrows or assertThrows for clarity and correctness.